starport/interface/cli/starport/cmd/network_chain_show.go

```diff
@@ -28,25 +39,62 @@ func networkChainShowHandler(cmd *cobra.Command, args []string) error {
 
 	chainID := args[0]
 
-	c, err := nb.ShowChain(context.Background(), chainID)
+	// Get flags
+	home, _, err := getHomeFlags(cmd)
 	if err != nil {
 		return err
 	}
-	chainyaml, err := yaml.Marshal(c)
+	showGenesis, err := cmd.Flags().GetBool(genesisFlag)
 	if err != nil {
 		return err
 	}
-
-	info, err := nb.LaunchInformation(context.Background(), chainID)
+	showPeers, err := cmd.Flags().GetBool(peersFlag)
 	if err != nil {
 		return err
 	}
-	infoyaml, err := yaml.Marshal(info)
+	if showGenesis && showPeers {
+		return fmt.Errorf("%s and %s flags cannot be set both", genesisFlag, peersFlag)
+	}
+
+	// Fetch launch information
+	info, err := nb.LaunchInformation(context.Background(), chainID)
 	if err != nil {
 		return err
 	}
 
-	fmt.Printf("\nChain:\n---\n%s\n\nLaunch Information:\n---\n%s", string(chainyaml), string(infoyaml))
+	switch {
+	case showGenesis:
+		// Generate the genesis in a temporary directory and show the content
+		tmpHome, err := nb.GenerateTemporaryGenesis(cmd.Context(), chainID, home, info)
```

Comment:
Why `home` needs to be configrable through flags?

Reply:
Because the home config of the chain can be in a custom directory. Every starport command has the `--home` flag to customize the home directory

Reply:
Yeah, but from what I understand from this; in order to these flags work, one of the join or create commands needs to run first?

Because `show` is about retrieving read-only chain info so, I think it shouldn't depend on those commands. Someone who isn't interested in creating or joining to a chain should be still able to use `show` on any existent chain.

So if show command does not depend on executing create/join commands first, a customizable `home` here is also not needed.

I think, `show` also shouldn't mutate any info inside a possibly existent home directory created by the create/join commands.

Reply:
I actually didn't make attention but we need to have run `create` or `join` command before and that's actually not very convenient for a potential new validator. Also,  `chain show`  is read-only in the sense that it has't any side effect on the launch workflow but in order to generate the genesis you still need to download the source, build it and initialize a config home.

I see two solution:

- The command will download the source, build and generate config home, everything in a temporary directory that will be then deleted in order to not have any side effect on the file system

- The command will download, build the binary using the same workflow as create/join/start commands (so the source will be in `.starport/spn-chain` and you can provide the `--home` flag)

The advantage of the second is that the download and building process is only performed once, for the first one even the coordinator would need to download and build each time the command is run

The disadvantage is that someone who just want to see the genesis of different chain would keep lot of files in their filesystem.

- A third solution is to do the first solution if the chain is not yet downloaded, and if the chain is already present in `.starport/spn-chain` we do the second solution (but then we must still be able to provide `--home` flag since it can be initialized in a custom home)

Reply:
@fadeev maybe you would have an opinion on this as well

Reply:
`show` command needs to init a chain first to apply changes to the genesis from SPN. It's perfectly fine if it creates a home directory (in the regular location) to do so. If there is something there already (rogue `~/.gaia`), then Starport should ask if it is allowed to nuke it. `--home` makes sense, if a user wants to see a genesis of a gaia-like chain, but doesn't want to remove their own `~/.gaia`.

Reply:
The third solution looks good!

Reply:
I think no matter which option we go with non of them will ever need to second download the blockchain repo if it is already cached under the `spn-chain` dir. Because caching is done by the `Init` function which is the entrypoint of `networkbuilder` service.
https://github.com/tendermint/starport/blob/74bb06ff6a005177128858781c21e9005fe9f0ad/starport/services/networkbuilder/networkbuilder.go#L155-L156

I think caching the blockchain's source code in the fs and where to locate its home directory are different problems. I believe having a customizable `--home` flag for the `show` command will just be confusing for developers and doesn't add anything in terms of functionality. 

Because what we do under the hood is to create a temprorary dir as a home dir and pass it to `chain.New` in all cases. Making the user choose a home dir or using a pre-existent one won't add anything. Re-execution of `appd init` with its siblings and building a genesis doesn't take noticeable time.  

As an additional improvement, I think we can cache blockchain binaries in the similar way that we do for their source codes. But I think it's not a blocker for this PR and once it's implemented, it'll be beneficial for the create/join commands as well.

Reply:
It is not that important to me to have it or not I was mainly following the Cosmos SDK command format that has the `--home` flag for all commands.

I can removing it if you think it would be confusing to the user